kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/DbAuthzGroupSqlDefault.java

```diff
@@ -1,511 +1,516 @@
+		sqlBuf.append("INNER JOIN SAKAI_REALM_FUNCTION SRF ON SRRF.FUNCTION_KEY = SRF.FUNCTION_KEY ");
+		sqlBuf.append("WHERE FUNCTION_NAME = ? and SRRG.ACTIVE = '1' and " + inClause + " ");
+	
+		return sqlBuf.toString();
+	}
+	
+	public String getSelectRealmRoleGroupUserCountSql(String inClause)
+	{
+		StringBuilder sqlBuf = new StringBuilder();
+		
+		sqlBuf.append("SELECT REALM_ID, COUNT(REALM_ID) ");
+		sqlBuf.append("FROM SAKAI_REALM SR INNER JOIN SAKAI_REALM_RL_GR SRRG ON SR.REALM_KEY = SRRG.REALM_KEY ");
+		sqlBuf.append("INNER JOIN SAKAI_REALM_RL_FN SRRF ON SRRF.ROLE_KEY = SRRG.ROLE_KEY AND SRRF.REALM_KEY = SR.REALM_KEY ");
+		sqlBuf.append("INNER JOIN SAKAI_REALM_FUNCTION SRF ON SRRF.FUNCTION_KEY = SRF.FUNCTION_KEY ");	
+		sqlBuf.append("WHERE FUNCTION_NAME = ? and SRRG.ACTIVE = '1' and " + inClause + " ");
+		sqlBuf.append("GROUP BY REALM_ID");
+	
+		return sqlBuf.toString();		
+	}
+	
+	public String getSelectRealmRoleNameSql()
+	{
+		return "select SRR.ROLE_NAME from SAKAI_REALM_RL_GR SRRG " + "inner join SAKAI_REALM SR on SRRG.REALM_KEY = SR.REALM_KEY "
+				+ "inner join SAKAI_REALM_ROLE SRR on SRRG.ROLE_KEY = SRR.ROLE_KEY "
+				+ "where SR.REALM_ID = ? and SRRG.USER_ID = ? and SRRG.ACTIVE = '1'";
+	}
+	
+	public String getSelectRealmRolesSql(String inClause)
+	{
+		return "select SR.REALM_ID, SRR.ROLE_NAME from SAKAI_REALM_RL_GR SRRG " + "inner join SAKAI_REALM SR on SRRG.REALM_KEY = SR.REALM_KEY "
+				+ "inner join SAKAI_REALM_ROLE SRR on SRRG.ROLE_KEY = SRR.ROLE_KEY "
+				+ "where SRRG.USER_ID = ? and SRRG.ACTIVE = '1' and " + inClause + " ";
+	}
+
+	public String getSelectRealmRoleSql()
+	{
+		return "select ROLE_NAME, ROLE_KEY from SAKAI_REALM_ROLE";
+	}
+
+	public String getSelectRealmRoleKeySql()
+	{
+		return "select ROLE_NAME, ROLE_KEY from SAKAI_REALM_ROLE where ROLE_NAME = ?";
+	}
+	
+	public String getSelectRealmSize()
+	{
+		return "select COUNT(REALM_KEY) from SAKAI_REALM_RL_GR where REALM_KEY = ?";
+	}
+
+	public String getSelectRealmUpdate()
+	{
+		return "select REALM_KEY from SAKAI_REALM where REALM_ID = ? FOR UPDATE";
+	}
+
+	public String getSelectRealmUserRoleSql(String inClause)
+	{
+		return "select SRRG.USER_ID, SRR.ROLE_NAME from SAKAI_REALM_RL_GR SRRG " + "inner join SAKAI_REALM SR on SRRG.REALM_KEY = SR.REALM_KEY "
+				+ "inner join SAKAI_REALM_ROLE SRR on SRRG.ROLE_KEY = SRR.ROLE_KEY " + "where SR.REALM_ID = ? and " + inClause
+				+ " and SRRG.ACTIVE = '1'";
+	}
+
+	public String getSelectRealmUsersInGroupsSql( String inClause)
+	{
+		return "select SRRG.USER_ID from SAKAI_REALM_RL_GR SRRG inner join SAKAI_REALM SR ON SRRG.REALM_KEY = SR.REALM_KEY where SRRG.ACTIVE = '1' and " + inClause;
+	}
+
+    public String getMaintainRolesSql() {
+        return "SELECT ROLE_NAME FROM SAKAI_REALM_ROLE WHERE ROLE_KEY IN (SELECT DISTINCT MAINTAIN_ROLE FROM SAKAI_REALM WHERE MAINTAIN_ROLE IS NOT NULL)";
+    }
+}
```

Comment:
Not sure why this file is being reported as being 100% changed, I only added 3 lines. I'll fix this one up tomorrow.